pgrx/src/datum/array.rs

```diff
@@ -722,23 +752,15 @@ impl<'a, T: FromDatum> Iterator for ArrayIntoIterator<'a, T> {
         Some(element)
     }
 
+    #[inline]
     fn size_hint(&self) -> (usize, Option<usize>) {
-        // If asking for size, it's not clear if they want "actual size"
```

Comment:
It's asking for actual size, e.g. returning `Some(None)` counts as returning an element.